fitbenchmarking/parsing/mantiddev_parser.py

```diff
@@ -0,0 +1,236 @@
+    """
+    Parser for a Mantid problem definition file.
+    """
+
+    def __init__(self, filename, options):
+        self._params_dict = None
+        self._mantid_function = None
+        self._jac = None
+        self._N_x = 0
+        self._cache_x = None
+        super().__init__(filename, options)
+
+    def _set_additional_info(self) -> None:
+        """
+        Sets any additional info for a fitting problem.
+        """
+        if self.fitting_problem.multifit:
+            self.fitting_problem.additional_info['mantid_ties'] \
+                = self._parse_ties()
+
+    def parse(self):
+
+        fp = super().parse()
+
+        self._set_jacobian(fp)
+        return fp
+
+    def _set_jacobian(self, fp) -> None:
+        """
+        Sometimes mantid will give the error
+        RuntimeError: Integration is not implemented for this function.
+        this try except tests if the error occurs and then only
+        assigns the jacobian if it passes.
+        :param fp: fitting problem
+        """
+        if self._is_multifit():
+            # currently cannot do Jacobian and multifit
+            return
+        # need to trim x data to the correct range for Jacobian
+        i0 = 0
+        iN = len(fp.data_x)
+        if fp.start_x:
+            i0 = np.argmax(fp.data_x >= fp.start_x)
+        if fp.end_x:
+            # returns a list of lists if more than one match,
+            # otherwise an int is returned
+            iN = np.where(fp.data_x <= fp.end_x)
+            if not isinstance(iN, int):
+                iN = iN[0][-1]
+
+        x_data = fp.data_x
+        x_data = x_data[i0:iN + 1]
+
+        # cache the x values for later
+        self._cache_x = FDV(x_data)
+        self._N_x = len(x_data)
+        self._jac = np.zeros((self._N_x, len(self._params_dict.keys())))
+
+        try:
+            _ = self._jacobian(x_data, self._params_dict.values())
+            fp.jacobian = self._jacobian
+        except RuntimeError:
+
+            return
+
+    def _jacobian(self, _x, *args):
+        # pylint: disable=unused-argument
+        """
+        Extracts the Jacobian from Mantid
+        WARNING: Gaussians are known to be incorrect
+
+        :param _x: the x values for the problem (assume they have not changed)
+        :param args: the input parameters for the fit
+        :return a matrix of the Jacobian
+        """
+        for param, key in zip(*args, self._params_dict.keys()):
+            self._mantid_function[key] = param
+        # get mantid Jacobian
+        J = self._mantid_function.functionDeriv(self._cache_x)
+        # set np Jacobian values
+        for i in range(self._N_x):
+            for j in range(len(self._params_dict.keys())):
+                self._jac[i, j] = J.get(i, j)
+        return self._jac
+
+    def _update_params(self, *p):
+        update_dict = dict(zip(self._params_dict.keys(), p))
+        self._params_dict.update(update_dict)
+        return self._params_dict
+
```

Comment:
Is this just left over from something you tried but didn't use in the end?
Or is it used somewhere I've missed?
```suggestion
```